controller/linodemachine_controller_helpers.go

```diff
@@ -712,3 +714,16 @@ func createInstance(ctx context.Context, logger logr.Logger, machineScope *scope
 	inst, err := machineScope.LinodeClient.CreateInstance(ctx, *createOpts)
 	return inst, ctr.RetryAfter(), err
 }
+
+// getVPCRefFromScope returns the appropriate VPC reference based on priority:
+// 1. Machine-level VPC reference
+// 2. Cluster-level VPC reference
+func getVPCRefFromScope(machineScope *scope.MachineScope) *corev1.ObjectReference {
+	if machineScope.LinodeMachine.Spec.VPCRef != nil {
+		return machineScope.LinodeMachine.Spec.VPCRef
+	}
+	if machineScope.LinodeCluster.Spec.VPCRef != nil {
+		return machineScope.LinodeCluster.Spec.VPCRef
+	}
+	return nil
```

Comment:
nit: could simplify this
```suggestion
	return machineScope.LinodeCluster.Spec.VPCRef
```